browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -0,0 +1,176 @@
+      profile_(Profile::FromWebUI(web_ui)),
+      weak_ptr_factory_(this) {
+  Profile* profile = Profile::FromWebUI(web_ui);
+  PrefService* prefs = profile->GetPrefs();
+  pref_change_registrar_ = std::make_unique<PrefChangeRegistrar>();
+  pref_change_registrar_->Init(prefs);
+  pref_change_registrar_->Add(
+      brave_rewards::prefs::kBraveRewardsEnabled,
+      base::Bind(&BraveRewardsInternalsUI::OnPreferenceChanged,
+                 weak_ptr_factory_.GetWeakPtr()));
+  web_ui->AddMessageHandler(std::make_unique<RewardsInternalsDOMHandler>(
+      weak_ptr_factory_.GetWeakPtr()));
+  rewards_service_ =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile_);
+  rewards_service_->AddObserver(this);
+  rewards_service_->GetRewardsInternalsInfo(
+      base::Bind(&BraveRewardsInternalsUI::OnGetRewardsInternalsInfo,
+                 weak_ptr_factory_.GetWeakPtr()));
+}
+
+BraveRewardsInternalsUI::~BraveRewardsInternalsUI() {
+  rewards_service_->RemoveObserver(this);
+}
+
+void BraveRewardsInternalsUI::CustomizeWebUIProperties(
+    content::RenderViewHost* render_view_host) {
+  DCHECK(IsSafeToSetWebUIProperties());
+
+  if (render_view_host) {
+    render_view_host->SetWebUIProperty("isRewardsEnabled",
+                                       BooleanToString(IsRewardsEnabled()));
+    if (internals_info_) {
+      render_view_host->SetWebUIProperty("walletPaymentId",
+                                         internals_info_->payment_id);
+      render_view_host->SetWebUIProperty(
+          "isKeyInfoSeedValid",
+          BooleanToString(internals_info_->is_key_info_seed_valid));
+
+      base::ListValue current_reconciles;
+      for (const auto& item : internals_info_->current_reconciles) {
+        auto current_reconcile_info = std::make_unique<base::DictionaryValue>();
+        current_reconcile_info->SetString("viewingId", item.second.viewing_id_);
+        current_reconcile_info->SetString("amount", item.second.amount_);
+        current_reconcile_info->SetInteger("retryStep",
+                                           item.second.retry_step_);
+        current_reconcile_info->SetInteger("retryLevel",
+                                           item.second.retry_level_);
+        current_reconciles.Append(std::move(current_reconcile_info));
+      }
+
+      std::string json;
+      if (base::JSONWriter::Write(current_reconciles, &json))
+        render_view_host->SetWebUIProperty("currentReconciles", json);
+    }
+  }
+}
+
+bool BraveRewardsInternalsUI::IsRewardsEnabled() const {
+  DCHECK(profile_);
+  return profile_->GetPrefs()->GetBoolean(
+      brave_rewards::prefs::kBraveRewardsEnabled);
+}
+
+void BraveRewardsInternalsUI::UpdateWebUIProperties() {
+  if (IsSafeToSetWebUIProperties()) {
+    CustomizeWebUIProperties(GetRenderViewHost());
+    web_ui()->CallJavascriptFunctionUnsafe(
+        "brave_rewards_internals.stateUpdated");
+  }
+}
+
+void BraveRewardsInternalsUI::OnWalletInitialized(
```

Comment:
this doesn't make sense to me. `OnWalletInitialized` may have been called long before the page is open and then it would never get the data. I think this also just needs to be dropped in favor of the method I outlined above.